exporter/exporterhelper/internal/base_exporter.go

```diff
@@ -204,13 +205,13 @@ func WithQueue(cfg queuebatch.Config) Option {
 // This option should be used with the new exporter helpers New[Traces|Metrics|Logs]RequestExporter.
 // Experimental: This API is at the early stage of development and may change without backward compatibility
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
-func WithQueueBatch(cfg queuebatch.Config, set queuebatch.Settings[request.Request]) Option {
+func WithQueueBatch(cfg configoptional.Optional[queuebatch.Config], set queuebatch.Settings[request.Request]) Option {
 	return func(o *BaseExporter) error {
-		if !cfg.Enabled {
+		if !cfg.HasValue() {
 			o.ExportFailureMessage += " Try enabling sending_queue to survive temporary failures."
 			return nil
```

Comment:
Is this `return nil` a problem? It will prevent modifying the settings in a case where HasValue() is false, but it's not clear why.